internal/to_tara/relayer.go

```diff
@@ -96,26 +97,28 @@ func (r *Relayer) Start(ctx context.Context) {
 
 	slot, err := r.beaconLightClient.Slot(nil)
 	if err != nil {
-		r.log.WithError(err).Fatal("Failed to get current slot from contract")
+		r.log.WithError(err).Panic("Failed to get current slot from contract")
 	}
 
 	r.currentContractSyncPeriod = common.GetPeriodFromSlot(int64(slot))
-	r.log.WithField("current period", r.currentContractSyncPeriod).Info("Beacon light client deployed, starting relayer")
+	r.log.WithField("BeaconPeriod", r.currentContractSyncPeriod).Info("Starting relayer")
 
 	r.processBridgeRoots()
 	r.applyStates()
 
-	go r.startEventProcessing(ctx)
-	go r.processNewBlocks(ctx)
-
 	r.checkAndFinalize()
 	r.checkAndUpdateNextSyncCommittee(r.currentContractSyncPeriod)
+
+	r.started = true
+
+	go r.startEventProcessing(ctx)
+	go r.processNewBlocks(ctx)
 }
 
-func (r *Relayer) Close() {
-	close(r.onFinalizedEpoch)
-	close(r.onFinalizedBlockNumber)
-	close(r.onSyncCommitteeUpdate)
+func (r *Relayer) Shutdown() {
+	for !r.started {
+		time.Sleep(100 * time.Millisecond)
+	}
 }
```

Comment:
```
Closing a channel indicates that no more values will be sent on it. This can be useful to communicate completion to the channel’s receivers.
```
I don't think that we need to notify, because we are stopping cycle that handling it when we are cancelling context 
```		
case <-ctx.Done():
	r.log.Info("Stopping new block processing")
```